README.md

```diff
@@ -1,10 +1,458 @@
+
+- `auxiliaryWorkers?: Array<AuxiliaryWorkerConfig>`
+
+  An optional array of auxiliary workers.
+  You can use [service bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/service-bindings/) to call auxiliary workers from your main (entry) Worker.
+  All requests are routed through your entry Worker.
+  During the build, each Worker is output to a separate subdirectory of `dist`.
+
+> [!NOTE]
+> When running `wrangler deploy`, only your main (entry) Worker will be deployed.
+> If using multiple Workers, each must be deployed individually.
+> You can inspect the `dist` directory and then run `wrangler deploy -c path-to-worker-output-config` for each.
+
+### `interface AuxiliaryWorkerConfig`
+
+- `configPath: string`
+
+  A required path to your Worker config file.
+
+- `viteEnvironment?: { name?: string }`
+
+  Optional Vite environment options.
+  By default, the environment name is the Worker name with `-` characters replaced with `_`.
+  Setting the name here will override this.
+
+## Worker environments
+
+A Worker config file may contain configuration for multiple [environments](https://developers.cloudflare.com/workers/wrangler/environments/).
+With the Cloudflare Vite plugin, you select an environment at dev or build time by providing the `CLOUDFLARE_ENV` environment variable.
+Consider the following example `wrangler.toml` file:
+
+```toml
+# wrangler.toml
+
+name = "my-worker"
+compatibility_date = "2024-12-30"
+main = "./src/index.ts"
+
+vars = { MY_VAR = "Top-level var" }
+
+[env.staging]
+vars = { MY_VAR = "Staging var" }
+
+[env.production]
+vars = { MY_VAR = "Production var" }
+```
+
+If you run `CLOUDFLARE_ENV=production vite build` then the output `wrangler.json` file generated by the build will be a flattened configuration for the 'production' environment.
+This combines [top-level only](https://developers.cloudflare.com/workers/wrangler/configuration/#top-level-only-keys), [inheritable](https://developers.cloudflare.com/workers/wrangler/configuration/#inheritable-keys), and [non-inheritable](https://developers.cloudflare.com/workers/wrangler/configuration/#non-inheritable-keys) keys.
+The value of `MY_VAR` will therefore be `'Production var'`.
+The name of the Worker will be `'my-worker-production'`
+This is because the environment name is automatically appended to the top-level Worker name.
+
+> [!NOTE]
+> The default Vite environment name for a Worker is always the top-level name.
+> This enables you to reference the Worker consistently in your Vite config when using multiple environments.
+
+Running `vite dev` or `vite build` without providing `CLOUDFLARE_ENV` will use the default top-level environment.
+The value of `MY_VAR` will therefore be `'Top-level var'`.
+As Worker environments are applied at dev and build time, specifying `CLOUDFLARE_ENV` when running `vite preview` or `wrangler deploy` will have no effect.
```

Comment:
Feels like this should be called out in a `!NOTE` block?

Reply:
There's already a `!NOTE` block directly above so it felt a bit weird having them next to each other. Any suggestions?

Reply:
I've fixed this by adding an additional paragraph in between - [886873f](https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/119/commits/886873f6fa7528620dc5cdc83d64ef689a18a5ae).

Let me know if that's OK.